vector-im/element-android
# Add metrics plugin to track device download keys task

## Type of change


## Content

- A plugin based approach is implemented so that clients using metrics sdk can integrate tracking of events/tasks.
- A new plugin `MetricsPlugin` is added which is implemented by `VectorMetricsPlugin`.
- Sentry tracking can track how many times the keys are being downloaded from server.

## Motivation and context

- In order to improve the crypto layer, we need to have some metrics to track and improve. `download_device_keys` is one of such metrics.

## Tests



- Enable analytics in Element-Android app.
- Try to open some conversations (rooms), especially the ones for which you don't have keys yet.
- You must see the count of `download_device_keys` increasing [here](https://sentry.tools.element.io/organizations/element/dashboard/4/?project=49).

## Tested devices

- OS version(s): Android 11 and 12

## Checklist



~- [ ] UI change has been tested on both light and dark themes~
~- [ ] Accessibility has been taken into account. See https://github.com/vector-im/element-android/blob/develop/CONTRIBUTING.md#accessibility~
~- [ ] Pull request includes screenshots or videos if containing UI changes~
~- [ ] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)~
~- [ ] If you have modified the screen flow, or added new screens to the application, you have updated the test [UiAllScreensSanityTest.allScreensTest()](https://github.com/vector-im/element-android/blob/main/vector/src/androidTest/java/im/vector/app/ui/UiAllScreensSanityTest.kt#L73)~
